packages/eslint-plugin/src/rules/no-misused-promises.ts

```diff
@@ -507,17 +510,41 @@ function voidFunctionParams(
       : subType.getConstructSignatures();
     for (const signature of signatures) {
       for (const [index, parameter] of signature.parameters.entries()) {
-        const type = checker.getTypeOfSymbolAtLocation(
+        const decl = parameter.getDeclarations()?.[0];
+        let type = checker.getTypeOfSymbolAtLocation(
           parameter,
           node.expression,
         );
+
+        const indices = [index];
+        // If this is a array 'rest' parameter, add all of the remaining parameter indices.
+        // Note - we currently do not support 'spread' arguments
+        if (
+          decl &&
+          ts.isParameter(decl) &&
+          decl.dotDotDotToken &&
+          checker.isArrayType(type) &&
```

Comment:
If I remove this line, the tests still pass (despite the type checking complaint on `checker.getTypeArguments(type)` below). Could you add test coverage for rest param type `any`, `any[]`, and `unknown[]`?

Reply:
I've added some additional rest parameter tests. It turns out that we weren't handling tuples at all - I've refactored the code to add support for tuple rest arguments, and added additional tests.